File: Java/Single-Number/Set.java
class Set {
    public int singleNumber(int[] nums) {
        //initialize a set
        Set<Integer> set = new HashSet();
        //for each int num in array nums
        for (int num : nums) {
            //check if set has not contained int num
            if(!set.contains(num)) {
                //add num to set
                set.add(num);
            }
            else {
                //otherwise, if num is already in the set, remove num from set
                set.remove(num);
            }
        }
        //return the first element in set
        return set.iterator().next();
    }
}

File: Python/Single-Number/Set.py
class Set:
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        #initialize a set
        num_set = set()
        #for each element in list nums
        for num in nums:
            #if duplicate/second element of the same num found
            if num in num_set:
                #remove from num_set
                num_set.remove(num)
            else:
                #otherwise, add element to num_set
                num_set.add(num)
        #return the only element left or the first element in num_set
        return num_set.pop()
